src/charts/brush.js

```diff
@@ -441,12 +442,39 @@ export default function module() {
      */
     function handleBrushStart() {
         const selection = event.selection;
+        let newSelection;
 
         if (!selection) {
             return;
         }
 
-        dispatcher.call('customBrushStart', this, selection.map(xScale.invert));
+        if (isLocked) {
+            const lockedSelectionSize = Math.floor(
+                xScale(new Date(dateRange[1])) - xScale(new Date(dateRange[0]))
+            );
+            const selectedRange = Math.floor(selection[1] - selection[0]);
+
+            if (
+                selectedRange < lockedSelectionSize ||
+                selectedRange > lockedSelectionSize
+            ) {
+                // We round values so we don't get into an infinite loop
+                newSelection = [
+                    Math.floor(selection[0]),
+                    Math.floor(selection[0]) + lockedSelectionSize,
+                ];
+                brush.move(chartBrush, newSelection);
+            } else {
+                newSelection = selection;
+            }
+        } else {
+            newSelection = selection;
```

Comment:
I see that:
```
 newSelection = selection;
```
assignment gets repeated. Maybe have a single case accounting for both instances?

Reply:
Yeah, I thought about this. However, then we would need to calculate lockedSelectionSize and selectedRange for all cases, and this event triggers A LOT. That's why I chose this not so-elegant looking way.

I think the main issue here is that the event should probably be debounced!